### ISSUE TITLE
BigQuery driver doesn't handle BYTES data type (and skips entire table)

### ISSUE THREAD

**Issue by tlrobinson** [metabase/metabase#4088]
Other data types we should ensure we handle: https://cloud.google.com/bigquery/data-formats

The `extraData` field in this case is type `BYTES`, and is being returned as ` {:name extraData, :base-type nil}`:

```
01-05 22:38:58 ERROR sync-database.introspect :: Unexpected error introspecting table schema: blocks #error {
 :cause Value does not match schema: {:fields #{{:base-type (not ("Valid field type" nil))}}}
 :data {:type :schema.core/error, :schema {:name (constrained Str Non-blank string), :schema (maybe (constrained Str Non-blank string)), :fields #{{:name (constrained Str Non-blank string), :base-type (pred Valid field type), #schema.core.OptionalKey{:k :special-type} (pred Valid field type), #schema.core.OptionalKey{:k :pk?} java.lang.Boolean, #schema.core.OptionalKey{:k :nested-fields} #{(recursive (var metabase.sync-database.interface/DescribeTableField))}, #schema.core.OptionalKey{:k :custom} {Any Any}}}}, :value {:schema nil, :name blocks, :fields #{{:name difficulty, :base-type :type/Text} {:name gasUsed, :base-type :type/Integer} {:name parentHash, :base-type :type/Text} {:name nonce, :base-type :type/Text} {:name transactionCount, :base-type :type/Integer} {:name hash, :base-type :type/Text} {:name size, :base-type :type/Integer} {:name gasLimit, :base-type :type/Integer} {:name totalDifficulty, :base-type :type/Text} {:name number, :base-type :type/Integer} {:name extraData, :base-type nil} {:name miner, :base-type :type/Text} {:name timestamp, :base-type :type/DateTime}}}, :error {:fields #{{:base-type (not (Valid field type nil))}}}}
 :via
 [{:type clojure.lang.ExceptionInfo
   :message Value does not match schema: {:fields #{{:base-type (not ("Valid field type" nil))}}}
   :data {:type :schema.core/error, :schema {:name (constrained Str Non-blank string), :schema (maybe (constrained Str Non-blank string)), :fields #{{:name (constrained Str Non-blank string), :base-type (pred Valid field type), #schema.core.OptionalKey{:k :special-type} (pred Valid field type), #schema.core.OptionalKey{:k :pk?} java.lang.Boolean, #schema.core.OptionalKey{:k :nested-fields} #{(recursive (var metabase.sync-database.interface/DescribeTableField))}, #schema.core.OptionalKey{:k :custom} {Any Any}}}}, :value {:schema nil, :name blocks, :fields #{{:name difficulty, :base-type :type/Text} {:name gasUsed, :base-type :type/Integer} {:name parentHash, :base-type :type/Text} {:name nonce, :base-type :type/Text} {:name transactionCount, :base-type :type/Integer} {:name hash, :base-type :type/Text} {:name size, :base-type :type/Integer} {:name gasLimit, :base-type :type/Integer} {:name totalDifficulty, :base-type :type/Text} {:name number, :base-type :type/Integer} {:name extraData, :base-type nil} {:name miner, :base-type :type/Text} {:name timestamp, :base-type :type/DateTime}}}, :error {:fields #{{:base-type (not (Valid field type nil))}}}}
   :at [schema.core$validator$fn__9060 invoke core.clj 155]}]
 :trace
 [[schema.core$validator$fn__9060 invoke core.clj 155]
  [schema.core$validate invokeStatic core.clj 164]
  [schema.core$validate invoke core.clj 159]
  [metabase.sync_database.introspect$introspect_tables_BANG_$fn__36253 invoke introspect.clj 152]
  [metabase.sync_database.introspect$introspect_tables_BANG_ invokeStatic introspect.clj 146]
  [metabase.sync_database.introspect$introspect_tables_BANG_ invoke introspect.clj 140]
  [metabase.sync_database.introspect$introspect_database_and_update_raw_tables_BANG_ invokeStatic introspect.clj 205]
  [metabase.sync_database.introspect$introspect_database_and_update_raw_tables_BANG_ invoke introspect.clj 196]
  [metabase.sync_database$sync_database_with_tracking_BANG_$fn__36544 invoke sync_database.clj 77]
  [metabase.sync_database$sync_database_with_tracking_BANG_ invokeStatic sync_database.clj 74]
  [metabase.sync_database$sync_database_with_tracking_BANG_ invoke sync_database.clj 68]
  [clojure.core$partial$fn__4763 invoke core.clj 2528]
  [metabase.driver$fn__20677 invokeStatic driver.clj 242]
  [metabase.driver$fn__20677 invoke driver.clj 242]
  [metabase.driver$eval20447$fn__20634$G__20414__20643 invoke driver.clj 43]
  [metabase.sync_database$sync_database_BANG_ invokeStatic sync_database.clj 43]
  [metabase.sync_database$sync_database_BANG_ doInvoke sync_database.clj 24]
  [clojure.lang.RestFn invoke RestFn.java 410]
  [metabase.events.sync_database$process_sync_database_event$fn__43439 invoke sync_database.clj 32]
  [clojure.core$binding_conveyor_fn$fn__4676 invoke core.clj 1938]
  [clojure.lang.AFn call AFn.java 18]
  [java.util.concurrent.FutureTask run FutureTask.java 266]
  [java.util.concurrent.ThreadPoolExecutor runWorker ThreadPoolExecutor.java 1142]
  [java.util.concurrent.ThreadPoolExecutor$Worker run ThreadPoolExecutor.java 617]
  [java.lang.Thread run Thread.java 745]]}
```

**Comment by camsaul** [comment]
BigQuery has been adding all sorts of new data types, `BYTES` didn't exist when I wrote the driver. I'll tweak it to fall back to `:type/*` (i.e. "Unknown") for types like these which is what the other drivers do

**Comment by camsaul** [comment]
Fixed by #4093
